src/main/java/seedu/address/logic/parser/SearchModeSearchCommandParser.java

```diff
@@ -0,0 +1,111 @@
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_TELEGRAM, PREFIX_ROLE);
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                PREFIX_TELEGRAM, PREFIX_ROLE);
+        //for each prefix, if present, parse the value and create a predicate...
+        // combine the predicates using AND
+        // return the search command with the predicate
+        // original predicate just show all
+        Set<Predicate<Person>> predicates = new HashSet<>();
+        // if a field is present, AND with the predicate for that field
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String name = argMultimap.getValue(PREFIX_NAME).get().trim();
+
+            String[] nameKeywords = name.split("\\s+");
+
+            Predicate<Person> namePred = new NameContainsKeywordsPredicate(
+                    Arrays.stream(nameKeywords).toList());
+            predicates.add(namePred);
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            String phone = argMultimap.getValue(PREFIX_PHONE).get().trim();
+            String[] phoneKeywords = phone.split("\\s+");
+
+            Predicate<Person> phonePred = new PhoneNumberContainsKeywordPredicate(
+                    Arrays.stream(phoneKeywords).toList());
+            predicates.add(phonePred);
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            String email = argMultimap.getValue(PREFIX_EMAIL).get().trim();
+            String[] emailKeywords = email.split("\\s+");
+
+            Predicate<Person> emailPred = new EmailContainsKeywordsPredicate(
+                    Arrays.stream(emailKeywords).toList());
+            predicates.add(emailPred);
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            String address = argMultimap.getValue(PREFIX_ADDRESS).get().trim();
+            String[] addressKeywords = address.split("\\s+");
+            Predicate<Person> addressPred = new AddressContainsKeywordsPredicate(
+                    Arrays.stream(addressKeywords).toList());
+
+            predicates.add(addressPred);
+        }
+        if (argMultimap.getValue(PREFIX_TELEGRAM).isPresent()) {
+            String telegram = argMultimap.getValue(PREFIX_TELEGRAM).get().trim();
+            String[] telegramKeywords = telegram.split("\\s+");
+            Predicate<Person> telegramPred = new TelegramContainsKeywordsPredicate(
+                    Arrays.stream(telegramKeywords).toList());
+            predicates.add(telegramPred);
+        }
+        //role have to use separate predicate
+        if (argMultimap.getValue(PREFIX_ROLE).isPresent()) {
+            String roles = argMultimap.getValue(PREFIX_ROLE).get().trim();
+            // map each word in String roles to a Role object
```

Comment:
Very long method, perhaps can refactor the code by abstracting each if block  into a separate method for better readability